### PR TITLE
bootstrap UI tests

### LINKED ISSUE #12: Bootstrap acceptance tests
Basic acceptance tests for ensuring the brute-force works as designed

#### Ensuring other users with the same IP can login
- ensure `user1` with `ip1` is blocked after x failed attempts
- ensure `user2` with `ip1` can still login

#### Ensuring that same user with different IP can login
- ensure `user1` with `ip1` is blocked after x failed attempts 
- ensure `user1` with `ip2` is allowed to login

Note - some things we might used during acceptance testing:
`X-Forwarded-For` Header would allow us to set an arbitrary IP in the scenario above

### PR DESCRIPTION
**Pull request by individual-it**
first step to fix #12
running the tests in drone will be next PR and then implementing API tests

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebUIBruteForceProtectionContext.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		$this->bruteForceProtectionPage = $bruteForceProtectionPage;
+		$this->loginPage = $loginPage;
+	}
+
+	/**
+	 * @When the administrator sets the bruteforceprotection settings using the webUI to:
+	 *
+	 * @param TableNode $table settings to be set
+	 * setting-value table without header
+	 * possible settings: treshold-time, fail-tolerance, ban-period
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIBruteForceProtectionContext.php]: s/treshold/threshold

File: tests/acceptance/features/lib/BlockedUserPage.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * ownCloud
+ *
+ * @author Paurakh Sharma Humagain <paurakh@jankaritech.com>
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BlockedUserPage.php]: Did you want to make Paurakh responsible for this code?

File: tests/acceptance/features/lib/BlockedUserPage.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ */
+
+namespace Page;
+
+use Behat\Mink\Session;
+
+/**
+ * Page that is presented to the blocked user when trying to login
+ * simmilar code is used in the user_management app for disabled user
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BlockedUserPage.php]: s/simmilar/similar

File: tests/acceptance/features/lib/BruteForceProtectionPage.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ *
+ */
+class BruteForceProtectionPage extends OwncloudPage {
+
+	/**
+	 *
+	 * @var string $path
+	 */
+	protected $path = '/index.php/settings/admin?sectionid=security';
+	private $tresholdTimeInputId = 'bfp-threshold-time';
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BruteForceProtectionPage.php]: s/tresholdTimeInputId/thresholdTimeInputId

global search for "treshold" and fix typo.

File: tests/acceptance/features/lib/BruteForceProtectionPage.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	private $failToloranceInputId = 'bfp-fail-tolerance';
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BruteForceProtectionPage.php]: s/failToloranceInputId/failToleranceInputId

File: tests/acceptance/features/lib/BruteForceProtectionPage.php [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	private $banPeriodInputId = 'bfp-ban-period';
+	private $saveButtonId = 'save-bfp-settings';
+	
+	/**
+	 * fills the input field of the setting with the given value
+	 *
+	 * @param string $setting treshold-time|fail-tolerance|ban-period
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BruteForceProtectionPage.php]: More treshold

File: tests/acceptance/features/lib/BruteForceProtectionPage.php [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	 * @return BruteForceProtectionPage
+	 */
+	public function setSetting($session, $setting, $value) {
+		$this->fillSettingInput($setting, $value);
+		$this->saveSettings($session);
+		return $this;
+	}
+
+	/**
+	 * sets multiple setting and saves them
```
Comment (phil-davis) [inline tests/acceptance/features/lib/BruteForceProtectionPage.php]: s/sets multiple setting/sets multiple settings

File: tests/acceptance/features/webUIBruteForceProtection/bruteforceprotection.feature [hunk 8]
```diff
@@ -0,0 +1,27 @@
+@webUI
+Feature: brute force protection
+
+As an administrator
+I want to be able to lock out users after multiple failed login attempts
+So that the server is protected against brute force password attacs
```
Comment (phil-davis) [inline tests/acceptance/features/webUIBruteForceProtection/bruteforceprotection.feature]: s/attacs/attacks

**Comment by individual-it** [comment]
@phil-davis typos fixed

File: tests/acceptance/features/bootstrap/WebUIBruteForceProtectionContext.php [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+	/**
+	 *
+	 * @var LoginPage
+	 */
+	private $loginPage;
+	/**
+	 * WebUIBruteForceProtectionContext constructor.
+	 *
+	 * @param OwncloudPage $owncloudPage
+	 * @param BruteForceProtectionPage $loginPage
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIBruteForceProtectionContext.php]: @individual-it this param is mixed up - should be 2 lines

Comment (individual-it) [reply tests/acceptance/features/bootstrap/WebUIBruteForceProtectionContext.php]: PR #20
php-cs-fixer did not shout at me, even the count of `@param` was not correct :-1:

**Pull request merged**